robot/web/web.go

```diff
@@ -498,6 +503,68 @@ func (svc *webService) runWeb(ctx context.Context, options weboptions.Options) (
 	return err
 }
 
+// Namer is used to get a resource name from incoming requests for countingfor request. Requests for
+// resources are expected to be a gRPC object that includes a `GetName` method.
+type Namer interface {
+	GetName() string
+}
+
+// RequestCounter maps string keys to atomic ints that get bumped on every incoming gRPC request for
+// components.
+type RequestCounter struct {
+	counts sync.Map
+}
+
+// UnaryInterceptor returns an incoming server interceptor that will pull method information and
+// optionally resource information to bump the request counters.
+func (mc *RequestCounter) UnaryInterceptor(
```

Comment:
nit: mc -> rc

Reply:
Oops, thanks. Was originally calling this "MethodCounter" and waffled.